### PR TITLE
dockerfile additions

### PR DESCRIPTION
**Pull request by nwb-adir**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,11 +24,16 @@ RUN wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor
 	chown root:root /etc/apt/sources.list.d/microsoft-prod.list
 
 RUN npm install -g serverless
-RUN apt-get update && apt-get install -y azure-cli dotnet-sdk-3.1 maven azure-functions-core-tools
+RUN npm install -g azure-functions-core-tools@3
+RUN apt-get update
+RUN apt install software-properties-common -y
+RUN add-apt-repository ppa:openjdk-r/ppa -y
+RUN apt-get install azure-cli git curl wget gcc zip unzip openjdk-8-jdk maven dotnet-sdk-3.1 python3 python3-pip -y --fix-missing
```
Comment (nwb-ilan) [inline Dockerfile]: Can you move as many of the libraries installed here as possible to the first apt-get on line 11.  
Also note to remove duplicates (curl, wget)

File: Dockerfile [hunk 2]
```diff
@@ -24,11 +24,16 @@ RUN wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor
 	chown root:root /etc/apt/sources.list.d/microsoft-prod.list
 
 RUN npm install -g serverless
-RUN apt-get update && apt-get install -y azure-cli dotnet-sdk-3.1 maven azure-functions-core-tools
+RUN npm install -g azure-functions-core-tools@3
+RUN apt-get update
+RUN apt install software-properties-common -y
```
Comment (nwb-ilan) [inline Dockerfile]: can we move this to line 11 as well?

File: Dockerfile [hunk 3]
```diff
@@ -24,11 +24,16 @@ RUN wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor
 	chown root:root /etc/apt/sources.list.d/microsoft-prod.list
 
 RUN npm install -g serverless
-RUN apt-get update && apt-get install -y azure-cli dotnet-sdk-3.1 maven azure-functions-core-tools
+RUN npm install -g azure-functions-core-tools@3
+RUN apt-get update
```
Comment (nwb-ilan) [inline Dockerfile]: Can be on the same line as `apt-get install`, best to minimize number of layers in Dockerfile.

File: Dockerfile [hunk 4]
```diff
@@ -24,11 +24,16 @@ RUN wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor
...
+RUN apt install software-properties-common -y
+RUN add-apt-repository ppa:openjdk-r/ppa -y
+RUN apt-get install azure-cli git curl wget gcc zip unzip openjdk-8-jdk maven dotnet-sdk-3.1 python3 python3-pip -y --fix-missing
 RUN mkdir /app
 
 COPY --from=builder /app/ /app
 WORKDIR /app
+RUN npm install
```
Comment (nwb-ilan) [inline Dockerfile]: maybe add `serverless` and `azure-functions-core-tools` to `package.json` instead of installing them individually?

Comment (nwb-adir) [reply Dockerfile]: they're installed globally, not at the module level

File: Dockerfile [hunk 5]
```diff
@@ -24,11 +24,16 @@ RUN wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor
 	chown root:root /etc/apt/sources.list.d/microsoft-prod.list
 
 RUN npm install -g serverless
-RUN apt-get update && apt-get install -y azure-cli dotnet-sdk-3.1 maven azure-functions-core-tools
+RUN npm install -g azure-functions-core-tools@3
```
Comment (nwb-ilan) [inline Dockerfile]: can we add this installation together with serverless?

**Comment by nwb-ilan** [comment]
@nwb-adir great.  
Does running faastest with docker work for all providers?

**Pull request merged**